test/integration/addons_test.go

```diff
@@ -743,3 +745,91 @@ func validateHeadlampAddon(ctx context.Context, t *testing.T, profile string) {
 		t.Fatalf("failed waiting for headlamp pod: %v", err)
 	}
 }
+
+// validateVolcanoAddon tests the Volcano addon, makes sure the Volcano is installed into cluster.
+func validateVolcanoAddon(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+
+	volcanoNamespace := "volcano-system"
+
+	client, err := kapi.Client(profile)
+	if err != nil {
+		t.Fatalf("failed to get Kubernetes client for %s: %v", profile, err)
+	}
+
+	// Wait for the volcano component installation to complete
+	start := time.Now()
+	var wg sync.WaitGroup
+	wg.Add(3)
+	go func() {
+		if err := kapi.WaitForDeploymentToStabilize(client, volcanoNamespace, "volcano-scheduler", Minutes(6)); err != nil {
+			t.Errorf("failed waiting for volcano-scheduler deployment to stabilize: %v", err)
+		}
+		t.Logf("volcano-scheduler stabilized in %s", time.Since(start))
```

Comment:
Can you move this log to an else block, currently, even if the deployment never comes up it logs that it stabilized which is not true.

Reply:
fixed